packages/sit-onyx/src/components/OnyxMenuItem/OnyxMenuItem.vue

```diff
@@ -0,0 +1,70 @@
+<script setup lang="ts">
+import { inject } from "vue";
+import OnyxListItem from "../OnyxListItem/OnyxListItem.vue";
+import { MENU_BUTTON_ITEM_INJECTION_KEY } from "../OnyxNavItem/future/types";
+import { type OnyxMenuItemProps } from "./types";
+
+const props = defineProps<OnyxMenuItemProps>();
+const emit = defineEmits<{
+  /**
+   * Emitted when the menu item is clicked (via click or keyboard).
+   */
+  click: [];
+}>();
+
+const menuButton = inject(MENU_BUTTON_ITEM_INJECTION_KEY);
+</script>
+<template>
+  <OnyxListItem
+    :selected="props.active"
+    v-bind="menuButton?.listItem"
+    :color="props.color"
+    :disabled="props.disabled"
+    class="onyx-menu-item"
+  >
+    <a
+      v-if="props.href"
+      class="onyx-future-nav-item__anchor"
+      v-bind="menuButton?.menuItem({ active: props.active })"
+      :href="props.href"
+      @click="emit('click')"
+    >
+      <slot></slot>
+    </a>
+    <button
+      v-else
+      class="onyx-menu-item__button onyx-list-item"
```

Comment:
For the `disabled` state we should set disabled="true" if it's a button and the aria-disabled="true" on the menuitem: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/menuitem_role#description